### PR TITLE
[cdc] misc. waivers including PAD reset issue

### PR DESCRIPTION
**Pull request by jeoongp**
W_ASYNC_RST_FLOPS :
 -spi_device and dmi jtag
W_RECON_GROUPS :
 -Signals driven by AST
 -Recon in RV PLIC

Signed-off-by: Joshua Park <jeoong@google.com>

### CODE REVIEW

File: hw/top_earlgrey/cdc/cdc_waivers.tcl [hunk 1]
```diff
@@ -50,6 +50,48 @@ set_rule_status -rule {W_GLITCH} -status {Waived}          \
     (GlitchOutput=~"*u_sync_1*")} \
   -comment {Waive PADs input goes into synchronizer}
 
+# Driving from PAD RESET to SPI TPM 
+set_rule_status -rule {W_ASYNC_RST_FLOPS} -status {Waived}                           \
+  -expression {(DrivingSignal=~"IO*") && (ReceivingFlop=~"*u_spi_device.*")} \
```
Comment (tjaychen) [inline hw/top_earlgrey/cdc/cdc_waivers.tcl:55]: do we need to make this one more specific? this seems to say any IO pad to anything in spi_device...

File: hw/top_earlgrey/cdc/cdc_waivers.tcl [hunk 2]
```diff
@@ -50,6 +50,48 @@ set_rule_status -rule {W_GLITCH} -status {Waived}          \
...
+  -comment {PAD RESET driving to SPI TPM}
+
+set_rule_status -rule {W_ASYNC_RST_FLOPS} -status {Waived}                           \
+  -expression {(DrivingSignal=~"top_earlgrey.u_pinmux_aon*") && (ReceivingFlop=~"*u_spi_device.*")} \
+  -comment {PAD RESET driving to SPI TPM}
+
+set_rule_status -rule {W_ASYNC_RST_FLOPS} -status {Waived}                           \
+  -expression {(DrivingSignal=~"top_earlgrey.u_rstmgr_aon*") && (ReceivingFlop=~"*u_spi_device.*")} \
+  -comment {PAD RESET driving to SPI TPM}
+
```
Comment (tjaychen) [inline hw/top_earlgrey/cdc/cdc_waivers.tcl:69]: similar to above, could this be made more specific? perhaps inside the spi_tpm module directly?

File: hw/top_earlgrey/cdc/cdc_waivers.tcl [hunk 3]
```diff
@@ -50,6 +50,48 @@ set_rule_status -rule {W_GLITCH} -status {Waived}          \
...
+# Driving from PAD PIN MUX RESET to DMI JTAG 
+set_rule_status -rule {W_ASYNC_RST_FLOPS} -status {Waived}                           \
+  -expression {(DrivingSignal=~"top_earlgrey.u_pinmux_aon*") && (ReceivingFlop=~"*u_dmi_jtag.*")} \
+  -comment {PAD RESET driving to dmi jtag}
+
+# dual port memory to SPI
+set_rule_status -rule {W_MASYNC} -status {Waived}                           \
+  -expression {(Driver =~ "top_earlgrey.u_spi_device.u_memory_2p*") && (ReceivingFlop=~"top_earlgrey.u_spi_device.*")} \
```
Comment (tjaychen) [inline hw/top_earlgrey/cdc/cdc_waivers.tcl:77]: i'm slightly worried about this one. If hypothetically we hooked up thigns up wrong.. ie port A is clk, port B is spi_clk, but then we actually hook up some spi logic to read port A.

Not sure you could actually make this more specific..might not be possible.

File: hw/top_earlgrey/cdc/cdc_waivers.tcl (hunk 1 above)
Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.tcl:55]: Right. I will place more specific paths instead of a common broad one.

Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.tcl]: It's updated with u_spi_device.u_spi_tpm* which is the common path for following signals.
Screen Shot 2022-07-25 at 2.51.50 PM

Comment (tjaychen) [reply hw/top_earlgrey/cdc/cdc_waivers.tcl]: thanks!

File: hw/top_earlgrey/cdc/cdc_waivers.tcl (hunk 3 above)
Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.tcl]: It's fixed to include more specific path. We are checking other issues related to dual port memory in parallel such as https://github.com/lowRISC/opentitan/issues/13834

File: hw/top_earlgrey/cdc/cdc_waivers.tcl (hunk 2 above)
Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.tcl:69]: I added spi_tpm path to be more specific. Then, I forgot to reply here.

**Review by tjaychen — APPROVED** [review]

**Pull request merged**